lib/dradis/plugins/qualys/was/importer.rb

```diff
@@ -38,20 +37,21 @@ def import(params={})
           return false
         end
 
-        logger.info { 'Global Summary information'}
+        logger.info { 'Global Summary information' }
 
         xml_global_summary = doc.at_xpath('WAS_SCAN_REPORT/SUMMARY/GLOBAL_SUMMARY')
         logger.info { 'Security Risk: ' + xml_global_summary.at_xpath('./SECURITY_RISK').text }
         logger.info { 'Vulnerabilities found: ' + xml_global_summary.at_xpath('./VULNERABILITY').text }
 
-        xml_webapp = doc.at_xpath('WAS_SCAN_REPORT/APPENDIX/WEBAPP')
+        xml_webapp =
+          doc.at_xpath('WAS_SCAN_REPORT/APPENDIX/WEBAPP | WAS_SCAN_REPORT/APPENDIX/WEB_APPLICATION')
         process_webapp(xml_webapp)
 
         doc.xpath('WAS_SCAN_REPORT/GLOSSARY/QID_LIST/QID').each do |xml_qid|
           process_issue(xml_qid)
         end
 
-        doc.xpath('WAS_SCAN_REPORT/RESULTS/VULNERABILITY_LIST/VULNERABILITY').each do |xml_vulnerability|
+        doc.xpath('WAS_SCAN_REPORT/RESULTS//VULNERABILITY_LIST/VULNERABILITY').each do |xml_vulnerability|
```

Comment:
This in particular worries me... It's a broad match that can come an bite us later.

Are we better off?:

1. Detecting the format
2. Having a constant that provides specific XPath search strings for each format in a Hash (e.g. `WAS_SCAN_REPORT/RESULTS/VULNERABILITY_LIST/VULNERABILITY` vs `WAS_SCAN_REPORT/RESULTS/WEB_APPLICATION/VULNERABILITY_LIST/VULNERABILITY`)
3. Using the format from #1 to choose the right set of XPath strings from #2

Reply:
Went with 1 and explicitly defined the format.